File: docs/python3/algorithmes/cmd_args_file_io.py
#  First programs in Python
#  Julien VALENTIN (2022)  julien.vlnt@gmail.com
#  Compute the sum of the two first arguments and write the result in the file,
#  its name being the third argument.
#  Run in a terminal : `python computer.py <m> <n> <ouputFile>`
import os
import sys
import time

# If the number is not equal to three : exit.
if( len(sys.argv) != 4 ):
    print("Usage : `python computer.py <m> <n> <outputFile>`")

else:
    # Get the two values for the sum.
    m = int(sys.argv[1])
    n = int(sys.argv[2])

    # Compute the sum.
    var = m + n

    # Write it in the specified file.
    try:
        with open(sys.argv[3], "w") as f:
            f.write(f"{var}")
    except:
        print("Error writing in the file !")

    # Wait for five seconds...
    time.sleep(5)

    # ... then delete the file.
    os.remove(sys.argv[3])

File: docs/fortran/algorithmes/cmd_args_file_io.f90
!  First programs in Fortran.
!  Julien VALENTIN (2022)  julien.vlnt@gmail.com
!  Compute the sum of the two first arguments and write the result in a file.
!  Usage : `computer.exe <m> <n> <ouputFile>`.
program computer

    ! Declare variables.
    character(len=8) :: charm, charn, filename
    integer :: m, n, var

    ! Test the number of arguments. 
    if(command_argument_count().ne.3) then
        print '("Usage : computer.exe <m> <n> <outputfile>")'  
        stop
    endif

    ! Get the arguments.
    call get_command_argument(1, charm)
    call get_command_argument(2, charn)
    call get_command_argument(3, filename)

    ! Translate charm and charn into integers.
    read(charm, *) m
    read(charn, *) n

    ! Compute the sum.
    var = m + n

    ! Write the result in the file.
    open(file=filename, unit=0)
    write(0, '(I0)') var
    
    ! Sleep for five seconds...
    call sleep(5)

    ! ... then delete the file !
    close(unit=0, status='delete')

end program